src/h5web/visualizations/shared/models.ts

```diff
@@ -23,20 +23,13 @@ export type Size = { width: number; height: number };
 
 export type Domain = [number, number];
 
-export interface IndexAxisConfig {
-  indexDomain: Domain;
-  showGrid?: boolean;
-  scaleType?: never; // invalid
-}
-
-export interface DataAxisConfig {
-  dataDomain: Domain;
+export interface AxisConfig {
+  isIndexAxis?: boolean;
+  domain: Domain;
   showGrid?: boolean;
   scaleType?: ScaleType;
 }
```

Comment:
`IndexAxisConfig` and `DataAxisConfig` were merged in a single general interface. I only kept a boolean `isIndexAxis` that will force the axis to display only integers (see the updated `AxisInfo`).